crates/sema/src/lib.rs

```diff
@@ -58,6 +60,12 @@ pub fn parse_and_resolve(pcx: ParsingContext<'_>) -> Result<()> {
         }
     }
 
+    if sess.ast_stats {
+        for source in sources.asts() {
+            stats::print_ast_stats(source, "AST STATS", "ast-stats");
+        }
+    }
```

Comment:
Should I add a test for this somewhere in the test suite? If so should it be testing from the CLI perspective, or just testing the `print_ast_stats` function?

Reply:
Yes, you can add a tests/ui/stats/ast.rs with a simple program and just pass `//@ compile-flags: -Zast-stats`

Reply:
Added.